cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -75,6 +75,17 @@ type systemdMountOptions struct {
 	Private bool
 	// Umount the mountpoint
 	Umount bool
+	// Overlayfs indicates an overlay filesystem.
+	Overlayfs bool
```

Comment:
I kind of want to make this into something like this:

```go
type systemdMountOptions struct {
   ...
   FileSystemOptions MountOptioner
}

type MountOptioner interface {
    MountOptions() []string
}

type OverlayMountOptions {
   LowerDir []string
   UpperDir string
   WorkDir  string
}

func (o *OverlayMountOptions) MountOptions() []string{
   return ...
}
```

This way we won't get a proliferation of lots of random fields in this structure while retaining clean type separation.

Opinions?

In addition, you can then perhaps attach MarshalText and UnmarshalText function to the same type and have a convenient place to know how to parse stuff.

Reply:
Given that this is quite internal I think we can do this another time.